File: modules/01-basic-syntax/python/types_inference.py
def takes_u32(x: int):
  print(f"u32: {x}")

def takes_i8(y: float):
  print(f"i8: {y}")

# Types inference
def main():
  print("------------------------\n")

  x = 10
  y = 20

  takes_u32(x)
  takes_i8(y)
  takes_u32(y)

if __name__ == "__main__":
  main()

File: modules/01-basic-syntax/rust/examples/types_inference.rs
fn takes_u32(x: u32) {
  println!("u32: {x}");
}

fn takes_i8(y: i8) {
  println!("i8: {y}");
}

// Types inference
fn main() {
  println!("------------------------\n");

  let x = 10;
  let y = 20;

  takes_u32(x);
  takes_i8(y);
  // takes_u32(y);
}
